### PR TITLE
Language server

### PR DESCRIPTION
**Pull request by metamorphix**
This PR adds a basic language server for Clue.

### Overview

For now, the language server will just highlight syntax errors. Whenever a Clue file is opened or saved, the Clue executable will be run on that file. If there is an error, the location will be highlighted and the error message will be shown.

There is a configuration to set the path to (or command to run) the Clue executable.

### Future improvements

- Currently, the error highlight extends to the last non-whitespace character. It would be nice if the compiler could provide the range of the error. Perhaps we could have a compiler option to output more detailed and structured information for the language server to consume.
- There will be a lot we can do once we get semantic, name, and type analysis into the compiler, such as go-to-definitions and hover tooltips.

### CODE REVIEW

File: package-lock.json
Comment (Maiori44) [inline package-lock.json:1]: shouldn't this be gitignored?

**Review by Maiori44 — CHANGES_REQUESTED** [review]
Everything seems good, besides the package-lock.json which iirc is meant to be gitignored?

File: package-lock.json
Comment (metamorphix) [reply package-lock.json:1]: It seems the usual practice is to include it in the repository (see [here](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5)).

Comment (Maiori44) [reply package-lock.json:1]: oh, my bad then

**Review by Maiori44 — APPROVED** [review]

**Pull request merged**